pkg/proxy/proxier.go

```diff
@@ -404,6 +466,14 @@ func iptablesInit(ipt iptables.Interface) error {
 	if _, err := ipt.EnsureChain(iptables.TableNAT, iptablesContainerPortalChain); err != nil {
 		return err
 	}
+
+	// TODO (comment from thockin): This is a little scary - if somehow these rules get out of order, this will consume
```

Comment:
One option would be to jump from chain 1 to chain 2, instead of doing -j chain1 / return / -j chain 2

```
-A INPUT -j CHAIN1

-A CHAIN1 -j CHAIN2
```

instead of

```
-A INPUT -j CHAIN1
-A INPUT -j CHAIN2
```

The downside is that then we must be careful that we already insert rules into CHAIN1 ahead of the jump to CHAIN2.

All-in, I think I prefer keeping the logic simpler and doing what you suggested in your note.  We could add a test that the jumps are in the correct order in kube-proxy.  And I think this will probably get revisited with the work to move to pure iptables.